src/python/pants/jvm/test/junit.py

```diff
@@ -128,6 +128,7 @@ async def setup_junit_for_target(
             *junit.args,
         ],
         input_digest=input_digest,
+        extra_jvm_options=junit.jvm_options,
```

Comment:
It looks like this is actually the first usage of the `extra_jvm_options` argument maybe?

In that case, I wonder if maybe the `JvmProcess` constructor could take an optional `JvmToolBase` and grab the option (and other values) in the future?

Reply:
@stuhood That's probably premature optimisation for now (`JvmProcess` is intended as a reasonably lightweight wrapper over `Process` to ensure that the JVM machinery is in place, rather than a fully-fledged abstraction).

I can think of a few patterns that might be better suited to supporting a more generic concept of specialised execution environment, i.e. how do we make the `JvmProcess -> Process` approach more generic?

Reply:
I did find it a bit premature as well to add another constructor since it would only replace one argument, although it would help to not forget to pass the `extra_jvm_options`.

Another alternative I thought of was on having in `JvmProcess` an optional `jdk_tool` field, so the per-tool JVM options could be picked up when doing the `JvmProcess -> Process` conversion. I would still retain the `extra_jvm_options` field in `JvmProcess` for other cases.

Does that sound like a reasonable approach?